src/libasr/asr_utils.h

```diff
@@ -4932,16 +4932,408 @@ static inline ASR::asr_t* make_ArrayPhysicalCast_t_util(Allocator &al, const Loc
     return ASR::make_ArrayPhysicalCast_t(al, a_loc, a_arg, a_old, a_new, a_type, a_value);
 }
 
-inline void flatten_ArrayConstant(Allocator& al, ASR::expr_t** a_args, size_t n_args, Vec<ASR::expr_t*> &new_args) {
+// inline void flatten_ArrayConstant(Allocator& al, void* data, size_t n_args, Vec<ASR::expr_t*> &new_args) {
+//     for (size_t i = 0; i < n_args; i++) {
+//         if (ASR::is_a<ASR::ArrayConstant_t>(*a_args[i])) {
+//             ASR::ArrayConstant_t* a_arg = ASR::down_cast<ASR::ArrayConstant_t>(a_args[i]);
+//             flatten_ArrayConstant(al, a_arg->m_args, a_arg->n_args, new_args);
+//         } else if (ASR::is_a<ASR::ArrayConstant_t>(*ASRUtils::expr_value(a_args[i]))) {
+//             ASR::ArrayConstant_t* a_arg = ASR::down_cast<ASR::ArrayConstant_t>(ASRUtils::expr_value(a_args[i]));
+//             flatten_ArrayConstant(al, a_arg->m_args, a_arg->n_args, new_args);
+//         } else {
+//             new_args.push_back(al, ASRUtils::expr_value(a_args[i]));
+//         }
+//     }
+// }
+
+inline void set_ArrayConstant_value(ASR::ArrayConstant_t* x, ASR::expr_t* value, int i) {
+    ASR::ttype_t* type = ASRUtils::type_get_past_array(ASRUtils::type_get_past_allocatable(ASRUtils::expr_type(value)));
+    std::cout<<"ASRUtils::type_to_str_python(type): "<<ASRUtils::type_to_str_python(type)<<"\n";
+    int kind = ASRUtils::extract_kind_from_ttype_t(type);
+
+    switch (type->type) {
+        case ASR::ttypeType::Integer: {
+            ASR::IntegerConstant_t* value_int = ASR::down_cast<ASR::IntegerConstant_t>(value);
+            if (kind == 1) {
+                *(((int8_t*)x->m_data) + i) = value_int->m_n;
```

Comment:
This should work:
```suggestion
                ((int8_t*)x->m_data)[i] = value_int->m_n;
```
And it is clearer what it means.